core/include/scipp/core/transform_common.h

```diff
@@ -62,6 +62,14 @@ static constexpr auto dimensionless_unit_check_return =
                  return units::one;
                }};
 
+template <typename Op> struct assign_op : Op {
+  template <typename Out, typename... In>
+  void operator()(Out &out, In &&...in) {
+    out = Op::operator()(std::forward<In>(in)...);
+  }
+};
+template <typename Op> assign_op(Op) -> assign_op<Op>;
```

Comment:
As commented above, should this be `assign_unary`?

Reply:
Maybe, I'm looking into generalising this right now. If that is not too difficult, we might as well.

Reply:
I'm not sure this is possible in a completely generic way. If the operation we wrap is an `overloaded`, we could capture its template parameters, inherit from anything that is not an `arg_list`, and augment the `arg_list`'s by adding the return type. (I think this can be deduced from the operation.) There could also be an extra template parameter to specify the type if we want to support the numpy style API mentioned above.
If the operation is not an `overloaded`, I don't see how we can filter out the `arg_list`.

In any case, this would be rather complicated and probably not worth the time. What do you think?

Reply:
> There could also be an extra template parameter to specify the type if we want to support the numpy style API mentioned above.

And to complicate things, we don't necessarily want to product space of listed types, e.g., if the unary op supports floats and ints, we may want to support float-precision conversion and int-precision conversion, but not float<->int conversions. The consequence would be that we need to specify type tuples, just as we did before with the separate `out_arg` kernels.

So I'd say it is not worth the time. Renamed to `assign_unary` for now, and hope that we don't want to add type conversion too soon, requiring essentially reverting what you did here?

Reply:
Renamed it to `assign_unary`. Let's use it for as long as we don't need type conversions in those operations.